clients/src/main/java/org/apache/kafka/common/requests/ListOffsetRequest.java

```diff
@@ -39,6 +39,7 @@
 import static org.apache.kafka.common.protocol.CommonFields.TOPIC_NAME;
 
 public class ListOffsetRequest extends AbstractRequest {
+    public static final long NEXT_LOCAL_TIMESTAMP = -3L;
```

Comment:
`EARLIEST_LOCAL_TIMESTAMP` looks to be a better name.

Reply:
The offset we are looking for is "the earliest offset that has not been sent to remote storage". It's not the earliest offset in local storage. After a segment is sent to remote storage, we will still keep it in local storage for a pretty long time

Reply:
I understand what this offset is for :) It looks like neither of the names are intuitive. We can address this later.